Preprod sprint1

TheoBoschetti opened the pull request.

TheoBoschetti committed:
Creation of components repo and Game component

TheoBoschetti committed:
Game add to app

TheoBoschetti committed:
Title component and img file add into App

TheoBoschetti committed:
UI components (EmptyDiv, ControlTab, css files) add into app

TheoBoschetti committed:
Map component added

TheoBoschetti committed:
Player component added

TheoBoschetti committed:
bug correction in map.css

TheoBoschetti committed:
map add, merging of pualine's and arnaut's work with ours

Gumatan committed:
Modification du composant Map pour qu'il utilise les sprites et rearangement/modifications de la structure pour permetre le merge avec l'ancien travail

Gumatan committed:
modification des sprites de link pour l'alignement

Oluane committed:
deleting useless tiles from tiles directory and modifing object of tiles in tilesMap.js

Oluane committed:
Merge branch 'Preprod-Sprint1' of https://github.com/WildCodeSchool/lille-0919-js-projet2-zeldaSnes into Preprod-Sprint1

TheoBoschetti committed:
initial player place changed

Oluane committed:
Merge branch 'Preprod-Sprint1' of https://github.com/WildCodeSchool/lille-0919-js-projet2-zeldaSnes into Preprod-Sprint1

TheoBoschetti committed:
Update of colors of UI

Gautier-Mille committed:
modifcation of tilesMap.js to declare collision

Gautier-Mille committed:
add collision management in Game.jsx

TheoBoschetti committed:
Upadte of the UI and the background

Gautier-Mille committed:
modifcation of collision on some tiles

TheoBoschetti committed:
Music button added in ControlTab compoenent

TheoBoschetti committed:
Merge branch 'collision' into Preprod-Sprint1

TheoBoschetti committed:
Collision system added to Game component

BastienSaulnier reviewed (dismissed):
Approved according to no bugs during tests, and reviewed by all the team

TheoBoschetti committed:
Component music created

BastienSaulnier reviewed (dismissed):
Everything looking okay and ready for the Music Component integration. ( With the others features previously validated )

LoicBrassart commented on frontend/src/Components/ControlsTab.css:
```diff
@@ -0,0 +1,43 @@
+  display: block;
+  width: 50%;
+  margin: 0 auto;
+  background-color: rgb(219, 219, 219);
```
Please be consistent across your units and methods uses in CSS
Typically, either use hexadecimal color codes, *or* `rgba` calls, but not a mix of both

LoicBrassart commented on frontend/src/Components/EmptyDiv.jsx:
```diff
@@ -0,0 +1,12 @@
+function EmptyDiv() {
+  return (
+    <div className="EmptyDiv">
+      <img src="sideLink.png" />
```
So... This EmptyDiv is not empty ?
So it's badly named ? :fearful:

LoicBrassart commented on frontend/src/Components/Player.css:
```diff
@@ -0,0 +1,5 @@
+.player {
+    position: absolute;
+    z-index: 1;
```
You can let this `z-index` if you can explain to me why you couldn't handle it in another way

LoicBrassart commented on frontend/src/Components/Player.jsx:
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import "./Player.css";
+
+// Affiche le player et calcul le mouvement a faire provenant de Map
```
Please leave your comments in English, please

LoicBrassart commented on frontend/src/Components/Title.jsx:
```diff
@@ -0,0 +1,13 @@
+function Title() {
+  return (
+    <h1>
+      <img alt="logo" src={ZeldaLogo} />
```
You should have at least some text in this `<h1>` tag, otherwise you'll be heavily penalised SEO-wise

LoicBrassart commented on frontend/src/Components/tilesMap.js:
```diff
@@ -0,0 +1,97 @@
+const tileNames = {                 /* v = vegetal, w = water, g = ground, b = buildings, o = objects */
+    'w001Z': 'water',
```
"Z" as in "Collizion" ? =D
It would be nice to find something more explicit. (At the *very* least, add this "Z" to the commentary just above!)

LoicBrassart commented on frontend/src/Components/ControlsTab.jsx:
```diff
@@ -0,0 +1,41 @@
+    };
+  }
+
+  getMusic() {
```
A `getSomething()` function should *never* update anything, it should only return the value of `something`
A better name for this function would be `toggleMusic()`

LoicBrassart commented on frontend/src/Components/ControlsTab.jsx:
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import Spacebar from "../img/Spacebar.svg";
```
`.svg` files are not composants -> their name should be in camelCase (no beginning upper case letter)

LoicBrassart commented on frontend/src/Components/ControlsTab.jsx:
```diff
@@ -0,0 +1,41 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      canMusic: false
```
Does my `ControlsTab` use the value of `canMusic` in any way ? It would be better to move both the state and the button to your `Music` component, since you'd be able to add sound to any website with only one component. As you wrote it, your `Music` is heavily linked to a `ControlsTab`

LoicBrassart commented on frontend/src/Components/Player.jsx:
```diff
@@ -0,0 +1,47 @@
+// Affiche le player et calcul le mouvement a faire provenant de Map
+
+class Player extends React.Component {
+  // Fonction qui récupere l'input du clavier pour afficher le bon asset et faire bouger le player
```
Same here

LoicBrassart commented on frontend/src/Components/Player.jsx:
```diff
@@ -0,0 +1,47 @@
+  // Fonction qui récupere l'input du clavier pour afficher le bon asset et faire bouger le player
+  getAsset() {
+    let direction;
+    switch (this.props.keyCode) {
```
- Instead of receiving a "keyCode", it'd be wiser to give a "direction" to this component
- If the "direction" props is one of these values ("left","back","right","front"), you could have skipped this function entirely and used in the `render()` method: `src={"link/${this.props.direction}.png"}`

LoicBrassart commented on frontend/src/Components/Player.jsx:
```diff
@@ -0,0 +1,47 @@
+      <div
+        className="player"
+        style={{
+          top: `${this.props.y * 32}px`,
```
Let's say I now want to use 64-pixels tiles ? It'd be better to move this "32" to your state and reuse it

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+
+      case 38:
+        event.preventDefault();
+        let newPosition2 = this.state.y - 1;
```
Wait, what ? What are these ~shitty~ variables, "positionN" ? Nope, you don't proceed like this !
You initialize *one* `newPosition` above the switch and you update it !

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+      case 37:
+        event.preventDefault();
+        let newPosition = this.state.x - 1;
+        if (
```
```
if(condition) {
    fck off
} else {
    do smthing
}
fck off
```
When I write it in pseudo-code, do you _see_ there is a problem in this algo ?
Please rewrite this section ;-)

BastienSaulnier replied:
Conditions reading simplified

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+    };
+  }
+
+  // Fonction qui écoute les input du clavier sur toutela fenetre
```
Comments in English

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+  }
+
+  // Fonction qui récupere l'input du clavier pour afficher le bon asset et faire bouger le player
+  getMouvment(event) {
```
getMovement()

BastienSaulnier replied:
Renamed to getMovement

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+
+  componentDidMount() {
+    window.onkeydown = event => {
+      if (this.state.canMoove) {
```
canMove

BastienSaulnier replied:
Renamed to canMove

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+        setTimeout(() => {
+          this.setState({ canMoove: true });
+        }, 120);
+        this.getMouvment(event);
```
- getMovement
- The function doesn't need the whole event, please send only mandatory data (here: event.keyCode)

BastienSaulnier replied:
Renamed to getMovement() + event.keyCode

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+      case 39:
+        event.preventDefault();
+        let newPosition3 = this.state.x + 1;
+        if (
```
Let's be clear: read only he condition, and tell me if it seems maintainable to you:
`newPosition3 > 19 || tilesMap[this.state.y][this.state.x + 1].includes("Z")`

You could:
- Create `x` and `y` variables to increase readibility
- Avoid numbered variables like `newPositionN`
- Avoid hard-coded values like "19" or "14"
- Avoid odd conditions like "Does this tile name contain a 'Z' ? Because that would mean it would trigger a collision !"

BastienSaulnier replied:
Let and Const declared for better code reading ( the 3 first points )

BastienSaulnier replied:
"Z" Specified as collision trigger in comment from tilesMap

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+        }
+        break;
+
+      case 40:
```
As long as we're speaking about weird conditions: what does "40" mean ? You should create a named constant ("LEFT" ?) in a `conf.js` file and use it.
Bonus point: it would make key customisation easier on the long run ;-)

BastienSaulnier replied:
Constants renamed

LoicBrassart commented on frontend/src/Components/Game.jsx:
```diff
@@ -0,0 +1,121 @@
+        }
+        break;
+      default:
+        return "";
```
This function returns an empty string if you entered an invalid input, and `undefined` in all other cases ? That doesn't seem very useful

LoicBrassart commented on frontend/src/Components/Map.jsx:
```diff
@@ -0,0 +1,43 @@
+import "./Map.css";
+
+class Map extends React.Component {
+  shouldComponentUpdate() {
```
The behavior will prevent your map from updating. It will prove a bad idea when you try and create a second map

TheoBoschetti replied:
Thank you for your feedback, for the next map we are going to load in the game, we are going to implement a boolean for geting true this method.

LoicBrassart commented on frontend/src/Components/Map.jsx:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  randomGrass() {
+    let randomNum = Math.floor(Math.random() * 40);
```
Avoid hard-coded values like 40, 6, 23, 21: when I list them, you can't remember what it's about can you ?
Please use aptly-named variables instead ;-)

LoicBrassart replied:
We talked about it, at least use some continuous values!

LoicBrassart commented on frontend/src/Components/Map.jsx:
```diff
@@ -0,0 +1,43 @@
+    let randomNum = Math.floor(Math.random() * 40);
+    if (randomNum === 6 || randomNum === 23) {
+      return "highGrass";
+    } else if (randomNum === 21) {
```
When you encounter a `return`, the rest of the code is *not read*. You don't need the `else` afterwards, you can just chain plain `if`, it'll be easier to read

BastienSaulnier replied:
Deleted "else if" replaced by "if"

LoicBrassart commented on frontend/src/Components/Map.jsx:
```diff
@@ -0,0 +1,43 @@
+    return (
+      <div className="map">
+        {tilesMap.map(row =>
+          row.map(column => {
```
I'm not working with a whole "column" in this case, but with one "tile" ("one column in a one-row-array", if you prefer !)

LoicBrassart reviewed (changes requested):
Generally speaking:
- Please use more CSS variables
- Comments in English
- Commit messages in English, with a beginning uppercase letter, and without typos

TheoBoschetti committed:
Useless Z-index removed in Player.css

TheoBoschetti committed:
Standardisation of all color rule in css

TheoBoschetti committed:
Rename of EmptyDiv component in LegtDiv

TheoBoschetti committed:
Commentary translate in english

TheoBoschetti committed:
h1 markup deleted from Title component

TheoBoschetti committed:
Commentary about Z in tilesMap component added (For tile that can't be crossed)

TheoBoschetti committed:
getMusic method renamed to toggleMusic in ControlsTab component

TheoBoschetti committed:
svg file rename without Uppercase

TheoBoschetti committed:
Music component changed and got out of the ControTab component

BastienSaulnier committed:
Renamed getMouvment() by getMovement()

BastienSaulnier committed:
Renamed canMoove() to canMove()

Oluane committed:
Modified the 'column' parameter in the map function in Map.jsx

BastienSaulnier committed:
Renamed to getMovement(event.Keycode) instead of getmovement(event)

BastienSaulnier committed:
Merge branch 'Preprod-Sprint1' of https://github.com/WildCodeSchool/lille-0919-js-projet2-zeldaSnes into Preprod-Sprint1

dderancourt committed:
fix des points suivants : initialisation de la variable NewPosition une seule fois dans la fonction getmovement(), le return à la fin de la fonction retournait une string vide, il ne retourne plus rien

BastienSaulnier committed:
Deleted "else if" replaced by "if"

dderancourt committed:
fix des points suivants : initialisation de la variable NewPosition une seule fois dans la fonction getmovement, le return retourne désormais rien et plus une string vide

Oluane committed:
comment added to explain randomGrass() numbers

TheoBoschetti committed:
Keycode in getMovmenrt method in Game composent changed to key for more clarity

TheoBoschetti committed:
keycode became key in getAsset method in Player component

TheoBoschetti committed:
Implementation of a state equal of the height and width of the assets

TheoBoschetti committed:
Merge branch 'Preprod-Sprint1' of https://github.com/WildCodeSchool/lille-0919-js-projet2-zeldaSnes into Preprod-Sprint1

BastienSaulnier committed:
"Needed before pull request revews"

 Merge branch 'Preprod-Sprint1' of https://github.com/WildCodeSchool/lille-0919-js-projet2-zeldaSnes into Preprod-Sprint1

BastienSaulnier committed:
Simplified reading of the getMovement() function and vars assignations

Gumatan reviewed (approved):
Pull request reviewed, good for Loic review.

LoicBrassart reviewed (approved):
Meh. Some errors you leftr, but I don't wanna cripple the project any llonger. 
I let it pass, but please try and fix these things before adding more code
